packages/bcp/src/cosmwasmconnection.ts

```diff
@@ -234,8 +234,7 @@ export class CosmWasmConnection implements BlockchainConnection {
     try {
       // tslint:disable-next-line: deprecation
       const response = await this.restClient.txsById(id);
-      const chainId = this.chainId();
-      return this.parseAndPopulateTxResponse(response, chainId);
+      return this.parseAndPopulateTxResponseSigned(response);
```

Comment:
Ahhh... so we only do the complex transform when getting exactly one tx, not for search.
This seems quite smart

Reply:
Jap. This interface is like this because of the Ethereum implementation where you search via scraper and get a single tx via RPC.